utils/obsproc/Smap2Ioda.h

```diff
@@ -70,14 +74,23 @@ namespace gdasapp {
 
       iodaVars.referenceDate_ = "seconds since 1970-01-01T00:00:00Z";
 
-      // calculate the seconds of Jan 1 of startyear since unix epoch
-      std::tm tm{};
-      // defaults are zero, Jan is zero
-      tm.tm_year = startYear - 1900;
-      tm.tm_mday = 1;
-      time_t unixStartYear = mktime(&tm);
+      int year = startYear;
+      int month = 1;
+      int day = 1;
 
-      int unixStartDay = unixStartYear + startDay * 86400;
+      // Bypassing validYYYYMMDD function within dateToJulian
+      if (year == -9999) {
+        year = 1970;
+      }
+
+      uint64_t julianDate = util::datefunctions::dateToJulian(year, month, day);
+
+      // Convert a date to Julian date
+      // Subtract Julian date for January 1, 1970 (epoch)
+      int daysSinceEpoch = julianDate - 2440588;
+
+      // Calculate seconds
+      int unixStartDay = (daysSinceEpoch + startDay) * 86400;
```

Comment:
I thought we were going away from the system's clock? If it's just a leftover variable name from the previous implementation, can we rename it to something without "unix".

Reply:
> I thought we were going away from the system's clock? If it's just a leftover variable name from the previous implementation, can we rename it to something without "unix".

OK, I just kept it what a previous left over